<?xml version="1.0"?>
<launch>
  <arg name="rviz" default="false" />
  <arg name="rtabmapviz" default="true" />

 <arg name="pi/2" value="1.5707963267948966" />
<arg name="optical_rotate" value="0 0 0 -$(arg pi/2) 0 -$(arg pi/2)" />
 
 <node pkg="tf" type="static_transform_publisher"  name="base_to_color"
      args="$(arg optical_rotate) iris_0/base_link iris_0/stereo_camera_frame 100" />

<!-- Run the ROS package stereo_image_proc -->
<group ns="/iris_0/stereo_camera" >
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc"/>

    <!-- Odometry -->
    <node pkg="rtabmap_ros" type="stereo_odometry" name="stereo_odometry" output="screen">
        <remap from="left/image_rect"       to="left/image_rect"/>
        <remap from="right/image_rect"      to="right/image_rect"/>
        <remap from="left/camera_info"      to="left/camera_info"/>
        <remap from="right/camera_info"     to="right/camera_info"/>

        <param name="frame_id" type="string" value="iris_0/base_link"/>
        <param name="odom_frame_id" type="string" value="odom"/>
        <param name="approx_sync"   type="bool" value="false"/>
        <param name="queue_size"    type="int" value="5"/>

        <param name="Odom/MinInliers" type="string" value="12"/>
        <param name="Odom/RoiRatios"  type="string" value="0.03 0.03 0.04 0.04"/>
    </node>     
</group>



  <group ns="rtabmap">
    <!-- Visual SLAM -->
    <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
      <param name="subscribe_stereo" type="bool" value="true"/>
     <param name="subscribe_depth"  type="bool" value="false"/>
     <param name="approx_sync"      type="bool" value="false"/>

     <remap from="left/image_rect"   to="/iris_0/stereo_camera/left/image_rect_color"/>
     <remap from="right/image_rect"  to="/iris_0/stereo_camera/right/image_rect"/>
     <remap from="left/camera_info"  to="/iris_0/stereo_camera/left/camera_info"/>
     <remap from="right/camera_info" to="/iris_0/stereo_camera/right/camera_info"/>

      <remap from="odom" to="/iris_0/stereo_camera/odom"/>

      <param name="frame_id"    type="string" value="iris_0/base_link"/>
      <param name="queue_size"  type="int"    value="30"/>
      <param name="approx_sync" type="bool"   value="false"/>

      <param name="Vis/MinInliers" type="string" value="12"/>
    </node>
  </group>

 
<!-- Visualisation RTAB-Map -->
 <group ns="rtabmap">  
  <!-- Visualisation RTAB-Map -->
  <node pkg="rtabmap_ros" type="rtabmapviz" name="rtabmapviz" args="-d $(find rtabmap_ros)/launch/config/rgbd_gui.ini" output="screen">
     <param name="subscribe_stereo"    type="bool"   value="true"/>
     <param name="subscribe_odom_info" type="bool"   value="true"/>
     <param name="queue_size"          type="int"    value="10"/>
     <param name="frame_id"            type="string" value="/iris_0/base_link"/>
     <remap from="left/image_rect"   to="/iris_0/stereo_camera/left/image_rect_color"/>
     <remap from="right/image_rect"  to="/iris_0/stereo_camera/right/image_rect"/>
     <remap from="left/camera_info"  to="/iris_0/stereo_camera/left/camera_info"/>
     <remap from="right/camera_info" to="/iris_0/stereo_camera/right/camera_info"/>
     <remap from="odom_info"         to="/iris_0/stereo_camera/odom_info"/>
     <remap from="odom"              to="/iris_0/stereo_camera/odom"/>
  </node>
</group>

</launch>
